README.md

```diff
@@ -31,6 +31,130 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use std::{error::Error, io};
+use tracing::{debug, error, info, span, warn, Level, instrument};
+
+#[instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(excitement = "yay!", "hello! I'm gonna shave a yak.");
+    if yak == 3 {
+        warn!("could not locate yak!");
+        // note that this is intended to demonstrate `tracing`'s features, not idiomatic
+        // error handling! in a library or application, you should consider returning
+        // a dedicated `YakError`. libraries like snafu or thiserror make this easy.
+        return Err(io::Error::new(io::ErrorKind::Other, "shaving yak failed!").into());
+    } else {
+        debug!("yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut yaks_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        debug!(yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(yak, error = error.as_ref(), "failed to shave yak!");
+        } else {
+            yaks_shaved += 1;
+        }
+        debug!(yaks_shaved);
+    }
+
+    yaks_shaved
+}
+```
```

Comment:
maybe these general use examples of instrumenting code with tracing should go in `## Usage` rather than `### In Libraries`, and that section could just contain the "Libraries shouldn't create subscribers" paragraph?

Reply:
Yes, I think that's a good idea.